fvwm/infostore.c

```diff
@@ -47,23 +45,19 @@ static void delete_metainfo(const char *);
 
 /* ---------------------------- exported variables (globals) --------------- */
 
-/* ---------------------------- local functions ---------------------------- */
-
-MetaInfo *new_metainfo(void)
-{
-	MetaInfo *mi;
-
-	mi = fxcalloc(1, sizeof *mi);
+struct meta_infos meta_info_q;
 
-	return mi;
-}
+/* ---------------------------- local functions ---------------------------- */
 
 void insert_metainfo(char *key, char *value)
 {
 	MetaInfo *mi;
 	MetaInfo *mi_new;
 
-	for (mi = mi_store; mi; mi = mi->next)
+	if (TAILQ_EMPTY(&meta_info_q))
+		TAILQ_INIT(&meta_info_q);
```

Comment:
@ThomasAdam I am a bit concerned about this line, what if something is added to the store but then removed and it is emptied, this means the `TAILQ_INIT` is run more than once and I'm unsure if this will cause memory leaks or not. I am considering adding a bool variable `do_initialize_store` to store the state of if it has been initialized or not to keep it from being initialized more than once, but unsure if that is needed or not.